src/goto-symex/symex_function_call.cpp

```diff
@@ -293,6 +294,26 @@ void goto_symext::symex_function_call_code(
       symex_assign(state, code);
     }
 
+    if(symex_config.havoc_undefined_functions)
+    {
+      // assign non det to function arguments if pointers
+      // are not const
+      for(const auto &arg : call.arguments())
+      {
+        if(
+          arg.type().id() == ID_pointer &&
+          !arg.type().subtype().get_bool(ID_C_constant) &&
+          arg.type().subtype().id() != ID_code)
+        {
+          exprt object = dereference_exprt(arg, arg.type().subtype());
+          exprt cleaned_object = clean_expr(object, state, true);
+          guard_expr_managert tmp;
+          const guard_exprt guard(true_exprt(), tmp);
```

Comment:
`state` already has a guard manager, why not use that one?

Reply:
now using that